### ISSUE TITLE
Character Based Model

### ISSUE THREAD

**Issue by david-waterworth** [huggingface/tokenizers#828]
I feel like it would be useful if we have a `model` that simply split the pre-tokenised words into characters. I hacked this by training a BPE model using just the printable ascii characters i.e.

```
tokenizer = tokenizers.Tokenizer(tokenizers.models.BPE(unk_token="<unk>"))
trainer = BpeTrainer(special_tokens=["<s>", "</s>", "<unk>", "<pad>", "<mask>"])
tokenizer.train_from_iterator([[c] for c in string.printable], trainer)
```

This seems to give me what I want, not 100% sure.

The reason this is useful is in many character-based applications I still want to be able to identify the original `words` using `tokenizer.encode().word_ids` and I still want to map special character sequences to single tokens (mask, pad, bos/eos etc) so the huggingface tokenizer is still really powerful even when working with character models.

I also still keep running into the issue that the normaliser runs before the pre-tokeniser - so it's impossible to generate lower case tokens that are split using any form of case sensitive rules since lower case normalisation is applied before rather than after pre-tokenisers. I created a custom pre-tokenizer but this adds runtime overhead plus extra code to serialise/deserialise the model so it would be really nice if this could be addressed.
